src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -85,6 +91,8 @@ public void init() throws Throwable {
 
             panel.frame().output(Localization.lang("Saving database") + "...");
             panel.setSaving(true);
+        } else if (filePath.isPresent()) {
+            saveAs(filePath.get().toFile());
```

Comment:
(i) Shouldn't the real working be done in `run`? - Maybe you just have to add some comment here why it is directly saved. Even though, the internal implementation seems to offload this using SPIN.
(ii) Has this something todo with the LOCAL/REMOTE state? Why not using this as checking condition?